src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupLoggingHelper.cs

```diff
@@ -88,9 +95,9 @@ internal static string GetParameterText(string prefix, string parameterName, ILi
                         sb.Append("\n");
                     }
 
-                    if (!Traits.Instance.EscapeHatches.LogTaskInputsVerbose && sb.Length >= parameterTextLimit)
+                    if (!logVerbose && (sb.Length >= parameterCharacterLimit || i > parameterLimit))
```

Comment:
Grab the first 0.9 * parameterLimit and the last 0.1 * parameterLimit.

Reply:
I don't think this is worth the additional complexity.